### ISSUE TITLE
Automated testing and local development UX for email login

### ISSUE THREAD

**Issue by pearkes** [usdigitalresponse/usdr-gost#1160]
Once #1143 is completed and there is a localstack setup (see https://github.com/usdigitalresponse/usdr-gost/pull/1157 which adds that). We should improve the email-based login system with the following:

* Make it possible for contributors to login without 1) configuring an SMTP server w/ node mailer and 2) adding themselves to the users seed: https://github.com/usdigitalresponse/usdr-gost/blob/10ffd7335b3bb482a40b35745d6d61225730c2aa/packages/server/seeds/dev/ref/users.js#L18-L139
* Write automated tests that validate SES was correctly called and that the login token contained in the email is valid for login for that user
* Add some sort-of UX helper (utilizing localstack) for local development to ease logging-in without bypassing the email flow

Per discussion in Slack cc/ @TylerHendrickson @as1729 

Note from @TylerHendrickson:

> The LocalStack SES service provides an [endpoint](https://docs.localstack.cloud/user-guide/aws/ses/#sent-emails) at GET /_aws/ses that can be used to retrieve sent SES messages as JSON.

**Comment by jakekreider** [comment]
I'm working on #653 and was planning on adding tests with localstack. Is that in progress elsewhere/should I hold off on that? I've got it part of the docker-compose setup and functional with SDK v3 but no tests yet.

**Comment by jakekreider** [comment]
(or, I can take on this ticket and roll it into the work for #653)

**Comment by pearkes** [comment]
@jakekreider I think #1157 adds localstack due to being blocked on that S3 stuff.

Do feel free to take this ticket on! Feel to ping me if you want to brainstorm the local dev ux.

**Comment by sanason** [comment]
Assigning myself this ticket, primarily to address this bullet point:

> Make it possible for contributors to login without 1) configuring an SMTP server w/ node mailer and 2) adding themselves to the users seed

because I found this step confusing during onboarding.

**Comment by TylerHendrickson** [comment]
I think this issue may have been addressed in https://github.com/usdigitalresponse/usdr-gost/pull/2540

@sanason Let me know if you agree and/or had anything else planned for this issue; if not I can close it out.

**Comment by sanason** [comment]
> I think this issue may have been addressed in #2540
> 
> @sanason Let me know if you agree and/or had anything else planned for this issue; if not I can close it out.

@TylerHendrickson Yeah, Jeff fixed all of the local development UX problems, both in [#2450](https://github.com/usdigitalresponse/usdr-gost/pull/2540) and by [adding Mailpit](https://github.com/usdigitalresponse/usdr-gost/pull/2893).

I did do some investigation related to the other suggestions in this ticket.

> Add some sort-of UX helper (utilizing localstack) for local development to ease logging-in without bypassing the email flow

I think the creator of the ticket was suggesting that we eliminate the SMTP email-sending pathway and use only the SES API email-sending pathway in all environments (mocking SES with Localstack in the local dev environment). The problem with this idea is that having a dev tool like Mailpit is really nice and I was unable to find any comparable "UX helper" that works with Localstack SES. The cloud/pro version of Localstack has a UI for viewing received emails but I didn't find anything good that was self-hosted and free.

> Write automated tests that validate SES was correctly called and that the login token contained in the email is valid for login for that user

Writing an e2e test for login still might be worth doing. I wrote a Cypress test that seemed to work, against a test db that I migrated and seeded manually, but I was defeated by the task of automating the initialization of the db. The way the repo is structured makes it difficult for the code in `e2e` to perform db migration and we don't have the backend service set up to handle its migrations itself on start-up.
